src/render/draw_sky.js

```diff
@@ -8,51 +8,101 @@ import Context from '../gl/context.js';
 import Texture from './texture.js';
 import Program from './program.js';
 import {smoothstep} from '../util/util.js';
+import {calculateGlobeMatrix} from './../geo/projection/globe.js';
 import type SourceCache from '../source/source_cache.js';
 import SkyboxGeometry from './skybox_geometry.js';
 import {skyboxUniformValues, skyboxGradientUniformValues} from './program/skybox_program.js';
+import {atmosphereUniformValues} from '../terrain/globe_raster_program.js';
 import {skyboxCaptureUniformValues} from './program/skybox_capture_program.js';
 import SkyLayer from '../style/style_layer/sky_style_layer.js';
 import type Painter from './painter.js';
 import {vec3, mat3, mat4} from 'gl-matrix';
+import browser from '../util/browser.js';
 import assert from 'assert';
 
 export default drawSky;
 
+function drawGlobeAtmosphere(painter: Painter) {
```

Comment:
TODO: Allow globe atmosphere to be rendered without preexisting sky layers.

Reply:
I've made a small PR for this: https://github.com/mapbox/mapbox-gl-js/pull/11344